### PR TITLE
Angular 12

### PR DESCRIPTION
**Pull request by raitisbe**
## Description

Upgrade to Angular 12
This PR also fixes some still unresolved bugs related to migration to OL 6.6 in hslayers-cesium and hslayers-material projects.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Dependency updates
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [x] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers-cesium/src/hscesium-layers.service.ts [hunk 1]
```diff
@@ -404,7 +406,7 @@ export class HsCesiumLayersService {
           this.viewer.dataSources
         ) {
           this.viewer.dataSources.add(<DataSource>cesium_layer);
-          if (getTitle(lyr) != 'Point clicked') {
+          if (getTitle(lyr as Layer<Source>) != 'Point clicked') {
```
Comment (DailisLangovskis) [inline projects/hslayers-cesium/src/hscesium-layers.service.ts:409]: @raitisbe Was done in PR #2102

**Review by DailisLangovskis — APPROVED** [review]
Besides, the HsCesium service changes, that are almost identical, everything look good! Maybe can You go over the same changes on other PR and decide which should stay?

File: projects/hslayers-cesium/src/hscesium-layers.service.ts (hunk 1 above)
Comment (raitisbe) [reply projects/hslayers-cesium/src/hscesium-layers.service.ts:409]: I thought I'd seen it somewhere :)

**Comment by raitisbe** [comment]
> Besides, the HsCesium service changes, that are almost identical, everything look good! Maybe can You go over the same changes on other PR and decide which should stay?

Sure, easier through solving rebase conflicts

**Pull request merged**